clang/test/CodeGen/constexpr-math.cpp

```diff
@@ -0,0 +1,51 @@
+// RUN: %clang_cc1 -x c++ -triple x86_64-unknown-unknown -std=c++23 \
+// RUN: -emit-llvm -o - %s | FileCheck %s
+
+// RUN %clang_cc1 -x c++ -triple x86_64-linux-gnu -emit-llvm -o - %s \
+// RUN -std=c++23
+
+#define INFINITY ((float)(1e+300 * 1e+300))
```

Comment:
You should also add tests in SemaCXX that the calculations work at compile time. e.g.,
```
static_assert(__builtin_fmin(15.24, 1.3) == 1.3, "");
```